From cd7e8ea7d082394b9721bd63508bd04c7dbc56a7 Mon Sep 17 00:00:00 2001
From: Vladimir Oltean <vladimir.oltean@nxp.com>
Date: Wed, 11 May 2022 12:50:16 +0300
Subject: [PATCH v5.19 3/3] net: dsa: introduce the dsa_cpu_ports() helper

Similar to dsa_user_ports() which retrieves a port mask of all user
ports, introduce dsa_cpu_ports() which retrieves the mask of all CPU
ports of a switch.

Signed-off-by: Vladimir Oltean <vladimir.oltean@nxp.com>
Signed-off-by: Jakub Kicinski <kuba@kernel.org>
(cherry picked from commit 465c3de42b5dcdf0fa8cf996a81de6ba0e8553a3)
---
 include/net/dsa.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

--- a/include/net/dsa.h
+++ b/include/net/dsa.h
@@ -516,6 +516,17 @@ static inline u32 dsa_user_ports(struct
 	return mask;
 }
 
+static inline u32 dsa_cpu_ports(struct dsa_switch *ds)
+{
+	struct dsa_port *cpu_dp;
+	u32 mask = 0;
+
+	dsa_switch_for_each_cpu_port(cpu_dp, ds)
+		mask |= BIT(cpu_dp->index);
+
+	return mask;
+}
+
 /* Return the local port used to reach an arbitrary switch device */
 static inline unsigned int dsa_routing_port(struct dsa_switch *ds, int device)
 {
